src/Homework4/BrandNewGreenhouse.java

```diff
@@ -1,25 +1,172 @@
 package Homework4;
 
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.InputMismatchException;
 import java.util.List;
 import java.util.ArrayList;
+import java.util.Scanner;
 
-public class BrandNewGreenhouse implements Greenhouse {
+public class BrandNewGreenhouse implements Greenhouse, Illuminated {
+    private static Scanner scanner = new Scanner(System.in);
     private List<Plant> plants;
     private double temperature;
+    private File file;
+
     public BrandNewGreenhouse() {
         plants = new ArrayList<>();
+        file = new File(".\\src\\Homework4\\MyGreenhouse.txt");
+    }
+
+    @Override
+    public void illuminateRed() {
```

Comment:
If this method has no implementation and now you know about exceptions it's a good idea to throw not implemented exception in case someone would try to call this method

Reply:
Added exceptions ExecutionControl.NotImplementedException